### PR TITLE
Ensure KVO works for USB config options

### PR DESCRIPTION
**Pull request by pmarkowsky**
This PR adds methods to SNTConfigurator to ensure KVO works for those options through out Santa.

### CODE REVIEW

File: Source/common/SNTConfigurator.m [hunk 1]
```diff
@@ -459,6 +459,23 @@ + (NSSet *)keyPathsForValuesAffectingEnableBadSignatureProtection {
...
 
++ (NSSet *)keyPathsForValuesAffectingBlockUSBMount {
+  return [self syncAndConfigStateSet];
+}
+
++ (NSSet *)keyPathsForValuesAffectingBannedUSBBlockMessage {
+  return [self syncAndConfigStateSet];
+}
+
++ (NSSet *)keyPathsForValuesAffectingRemountUSBBlockMode {
```
Comment (russellhancox) [inline Source/common/SNTConfigurator.m:470]: This should be `keyPathsForValuesAffectingRemountUSBMode`

File: Source/common/SNTConfigurator.m [hunk 2]
```diff
@@ -459,6 +459,23 @@ + (NSSet *)keyPathsForValuesAffectingEnableBadSignatureProtection {
...
+  return [self syncAndConfigStateSet];
+}
+
++ (NSSet *)keyPathsForValuesAffectingRemountUSBBlockMessage {
+  return [self syncAndConfigStateSet];
+}
+
+
```
Comment (russellhancox) [inline Source/common/SNTConfigurator.m:478]: I also see a property `usbBlockMessage`, we should probably have a `keyPathsForValuesAffectingUsbBlockMessage` too

File: Source/common/SNTConfigurator.m (hunk 1 above)
Comment (tburgin) [inline Source/common/SNTConfigurator.m:470]: RemountUSBMode ?

File: Source/common/SNTConfigurator.m [hunk 3]
```diff
@@ -459,6 +459,23 @@ + (NSSet *)keyPathsForValuesAffectingEnableBadSignatureProtection {
   return [self configStateSet];
 }
 
++ (NSSet *)keyPathsForValuesAffectingBlockUSBMount {
+  return [self syncAndConfigStateSet];
+}
+
++ (NSSet *)keyPathsForValuesAffectingBannedUSBBlockMessage {
+  return [self syncAndConfigStateSet];
```
Comment (russellhancox) [inline Source/common/SNTConfigurator.m:467]: This one only comes from the config

File: Source/common/SNTConfigurator.m [hunk 4]
```diff
@@ -459,6 +459,23 @@ + (NSSet *)keyPathsForValuesAffectingEnableBadSignatureProtection {
...
+}
+
++ (NSSet *)keyPathsForValuesAffectingRemountUSBBlockMode {
+  return [self syncAndConfigStateSet];
```
Comment (russellhancox) [inline Source/common/SNTConfigurator.m:471]: This one only comes from the config

**Review by russellhancox — CHANGES_REQUESTED** [review]

File: Source/common/SNTConfigurator.m (hunk 1 above)
Comment (pmarkowsky) [reply Source/common/SNTConfigurator.m]: Fixed.

Comment (pmarkowsky) [reply Source/common/SNTConfigurator.m]: Fixed.

File: Source/common/SNTConfigurator.m (hunk 2 above)
Comment (russellhancox) [reply Source/common/SNTConfigurator.m:481]: It needs to be "AffectingUsbBlockMessage" - the property is `usbBlockMessage`

**Review by russellhancox — APPROVED** [review]

**Pull request merged**